dataset/base_data_module.py

```diff
@@ -0,0 +1,62 @@
+from os.path import exists
+from os.path import join
+from typing import Any
+
+import torch
+from pytorch_lightning import LightningDataModule
+from torch.nn.utils.rnn import pad_sequence
+from torch.utils.data import random_split, DataLoader
+
+from .contrastive_dataset import ContrastiveDataset
+from .download import load_dataset
+from .text_dataset import TextDataset
+
+text_datasets = ["poj_104"]
+
+
+class BaseDataModule(LightningDataModule):
+    def __init__(self, dataset_name: str, batch_size: int, is_test: bool = False):
+        super().__init__()
+        self.dataset_name = dataset_name
+        self.dataset_path = join("data", dataset_name)
+        self.batch_size = batch_size
+        self.is_test = is_test
+
+    def prepare_data(self):
+        if not exists(self.dataset_path):
+            load_dataset(self.dataset_name)
+
+        if self.dataset_name in text_datasets:
+            self.clf_dataset = TextDataset(dataset_path=self.dataset_path, is_test=self.is_test)
+        else:
+            raise NotImplementedError("Non-text datasets are currently not available")
+
+        self.dataset = ContrastiveDataset(clf_dataset=self.clf_dataset)
+
+    def setup(self, stage: str = None):
+        len_train = int(0.6 * len(self.dataset))
+        len_test = int(0.2 * len(self.dataset))
+        len_val = len(self.dataset) - len_test - len_train
+        self.train, self.val, self.test = random_split(self.dataset, [len_train, len_val, len_test])
```

Comment:
Should you fix here random seed?
It seems unnecessary since this is a torch function. But it's better to recheck.
Also, what behavior you expected if a dataset is already split into holdouts?

Reply:
There is no seed arg for `random_split`, I moved the logic of splitting the dataset to the `load_data` function